src/accounts_refactor.rs

```diff
@@ -0,0 +1,339 @@
+        event: LedgerEvent<L>,
+        source: EventSource,
+        records_commitment: MerkleCommitment,
+    ) -> Result<AccountEditor<'a, L, Key>, KeystoreError<L>> {
+        let mut scan = match self.account.scan.take() {
+            Some(scan) => scan,
+            None => return Err(KeystoreError::ScanNotFound),
+        };
+        scan.handle_event(event, source);
+        // Check if the scan is complete.
+        if let Err(scan) = scan.finalize(records_commitment) {
+            self.account.scan = Some(scan);
+        }
+        Ok(self)
+    }
+
+    /// Save the account to the store.
+    ///
+    /// Returns the stored account.
+    pub fn save(&mut self) -> Result<Account<L, Key>, KeystoreError<L>> {
+        self.store.store(&self.account.pub_key(), &self.account)?;
+        Ok(self.account.clone())
+    }
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> Deref
+    for AccountEditor<'_, L, Key>
+{
+    type Target = Account<L, Key>;
+
+    fn deref(&self) -> &Account<L, Key> {
+        &self.account
+    }
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> DerefMut
+    for AccountEditor<'_, L, Key>
+{
+    fn deref_mut(&mut self) -> &mut Account<L, Key> {
+        &mut self.account
+    }
+}
+
+/// Accounts stored in an accounts store.
+pub struct Accounts<L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> {
+    /// A key-value store for accounts.
+    store: AccountsStore<L, Key, Key::PubKey>,
+}
+
+impl<L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> Accounts<L, Key> {
+    #![allow(dead_code)]
+
+    /// Load an accounts store.
+    pub fn new(store: AccountsStore<L, Key, Key::PubKey>) -> Result<Self, KeystoreError<L>> {
+        Ok(Self { store })
+    }
+
+    /// Iterate through the accounts.
+    pub fn iter(&self) -> impl Iterator<Item = Account<L, Key>> + '_ {
+        self.store.iter().cloned()
+    }
+
+    /// Get the account by the public key from the store.
+    pub fn get(&self, pub_key: &Key::PubKey) -> Result<Account<L, Key>, KeystoreError<L>> {
+        Ok(self.store.load(pub_key)?)
+    }
+
+    /// Get a mutable account editor by the public key from the store.
+    pub fn get_mut(
+        &mut self,
+        pub_key: &Key::PubKey,
+    ) -> Result<AccountEditor<L, Key>, KeystoreError<L>> {
+        let account = self.get(pub_key)?;
+        Ok(AccountEditor::new(&mut self.store, account))
+    }
+
+    /// Commit the store version.
+    pub fn commit(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.commit_version()?)
+    }
+
+    /// Revert the store version.
+    pub fn revert(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.revert_version()?)
+    }
+
+    /// Create an account.
+    ///
+    /// Returns the editor for the created account.
+    pub fn create(
+        &mut self,
+        key: Key,
+        description: String,
+        assets: Vec<AssetCode>,
+        records: Vec<RecordInfo>,
```

Comment:
It feels wrong that these are required to create an account. They are computed from other data (e.g. which records are currently in the wallet) and are going to change dynamically anyway. This is making me feel a bit stronger that `assets` and `records` should not be fields in the `Accounts` model but rather should be computed data at the controller level